controllers/keycloakclient/chain/put_client_scope.go

```diff
@@ -27,17 +27,50 @@ func (el *PutClientScope) Serve(ctx context.Context, keycloakClient *keycloakApi
 }
 
 func (el *PutClientScope) putClientScope(ctx context.Context, keycloakClient *keycloakApi.KeycloakClient, realmName string) error {
+	if err := el.putDefaultClientScope(ctx, keycloakClient, realmName); err != nil {
+		return err
+	}
+
+	if err := el.putOptionalClientScope(ctx, keycloakClient, realmName); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (el *PutClientScope) putDefaultClientScope(ctx context.Context, keycloakClient *keycloakApi.KeycloakClient, realmName string) error {
 	kCloakSpec := keycloakClient.Spec
+
 	if len(kCloakSpec.DefaultClientScopes) == 0 {
 		return nil
 	}
 
-	scopes, err := el.keycloakApiClient.GetClientScopesByNames(ctx, realmName, kCloakSpec.DefaultClientScopes)
+	defaultScopes, err := el.keycloakApiClient.GetClientScopesByNames(ctx, realmName, kCloakSpec.DefaultClientScopes)
+	if err != nil {
+		return errors.Wrap(err, "error during GetClientScope")
+	}
+
+	err = el.keycloakApiClient.AddDefaultScopeToClient(ctx, realmName, kCloakSpec.ClientId, defaultScopes)
```

Comment:
These two "helper functions" could be refactored into one "DRY"-er function by passing the scopes and API functions as parameters, but that looked a bit dirty to me, so I'm not sure if that's better/more maintainable.